GPT 3.5 Turbo

mangeloni30 opened the pull request:
**AI Data Dashboard**

- Using gpt-3.5-turbo model

mangeloni30 committed:
feat: add gpt-3.5-turbo

mangeloni30 committed:
feat: add question field for gpt-3.5-turbo

apresmoi commented on openai/template.ts:
```diff
@@ -47,5 +47,5 @@ if(row['value']) {
 export const promptGPT35TurboTemplate = `
   ${promptTemplate}
-  Reply in json format only but even the jasvascript function should be in json format, just that and without non-whitespace character
+  Reply in json format only but even the javascript function should be in json format, just that and without whitespace character
```
Let's try to iterate this part in a more concise way, something like:

```
- Reply only in JSON format
- Javascript functions should be in string format, without any whitespace characters```

of course we should check if it works

apresmoi commented on openai/completions.ts:
```diff
@@ -43,26 +54,32 @@ AI: ${i.reply || ""}`
   interactions: ChatInteraction[],
-  options: { apikey: string }
+  options: { apikey: string },
+  isGpt35Turbo = false
```
Instead of adding this boolean, we could add the option `model: Model` to the `options` object.

```
type Model = "gpt-3.5-turbo" | "text-davinci-003"
```

apresmoi commented on openai/completions.ts:
```diff
@@ -5,27 +5,38 @@ type ChatInteraction = {
 
 export async function queryCompletions(
```
This function should be splitted into two: `queryComplations` and `queryCompletionsChat`.

apresmoi commented on openai/completions.ts:
```diff
@@ -43,26 +54,32 @@ AI: ${i.reply || ""}`
+  options: { apikey: string },
+  isGpt35Turbo = false
 ): Promise<ChatInteraction[]> {
   const completion = await queryCompletions(
```
Here you should call `queryCompletions`/`queryCompletionsChat` depending on what model you are using. If you are using 3.5-turbo you are using chat, etc.

apresmoi reviewed (changes requested):
Hey ! Thanks for the PR.

We need to make some changes in order to add this.

- First we should try to split the logic between the models. They behave differently, one returns texts and the other handles objects.

- For now let's keep it simple and let's keep the function `generatePrompt` returning a string for all models. So we can keep using the same UI for the prompt view.

- We should add a select input in the config panel (where you configure your API KEY) to be able to switch between models. This parameter should be saved and handled in the same way that the API KEY is being handled.

- For the functions handling the "chat/completions" endpoint, we shouldn't be doing any spliiting on the responses like we do for the completions, since the texts are already split into roles.

Let me know if you have any doubts!

mangeloni30 committed:
feat: add functions for each model

mangeloni30 commented:
> Hey ! Thanks for the PR.
> 
> We need to make some changes in order to add this.
> 
> * First we should try to split the logic between the models. They behave differently, one returns texts and the other handles objects.
> * For now let's keep it simple and let's keep the function `generatePrompt` returning a string for all models. So we can keep using the same UI for the prompt view.
> * We should add a select input in the config panel (where you configure your API KEY) to be able to switch between models. This parameter should be saved and handled in the same way that the API KEY is being handled.
> * For the functions handling the "chat/completions" endpoint, we shouldn't be doing any spliiting on the responses like we do for the completions, since the texts are already split into roles.
> 
> Let me know if you have any doubts!



> Hey ! Thanks for the PR.
> 
> We need to make some changes in order to add this.
> 
> * First we should try to split the logic between the models. They behave differently, one returns texts and the other handles objects.
> * For now let's keep it simple and let's keep the function `generatePrompt` returning a string for all models. So we can keep using the same UI for the prompt view.
> * We should add a select input in the config panel (where you configure your API KEY) to be able to switch between models. This parameter should be saved and handled in the same way that the API KEY is being handled.
> * For the functions handling the "chat/completions" endpoint, we shouldn't be doing any spliiting on the responses like we do for the completions, since the texts are already split into roles.
> 
> Let me know if you have any doubts!

Hi, I have pushed new changes regarding with your suggestions. Can you pls take a look on that?

apresmoi commented on models/index.ts:
```diff
@@ -0,0 +1,20 @@
+  "gpt-3.5-turbo": promptGPT35TurboTemplate
+};
+
+export const getGPTModel: GPTModel = {
```
I would change this for a switch(model) in the openai/completions.ts file.

apresmoi commented on models/index.ts:
```diff
@@ -0,0 +1,20 @@
+
+export const getModelOptions = () => ["text-davinci-003", "gpt-3.5-turbo"];
+
+export const getPromptModel: PromptModel = {
```
Also this, let's make it a switch in the other file.

apresmoi reviewed (commented):
Hey ! 

I added you a review, let's change those for a switch in the other file.

mangeloni30 committed:
feat: modify the structure

mangeloni30 commented:
> Hey !
> 
> I added you a review, let's change those for a switch in the other file.

Hey JC, I have pushed more changes, this time the structure is different. Let me know how does this look to you.

mangeloni30 committed:
feat: change files names and structure

mangeloni30 commented:
Hi @apresmoi, I just added the following changes. 

- "model" folder is inside of "openai" folder
- Rename `textDavinci003Completions` function to `queryTextDavinci003Completions`
- Rename `gpt35TurboCompletions` function to `queryGpt35TurboCompletions`
- Rename Dropdown file name to SelectInput
- Each query function (text-davinci-003 - gpt-3.5-turbo) is returning the chat object directly so in `queryCompletionsChat` function I have added a `switch` statement to return the result directly.

apresmoi commented:
Hey thanks again for the PR and corrections. I tried it out and it seems to work as spected.

Merging this.
